src/internal/components/chart-legend/index.tsx

```diff
@@ -120,44 +125,45 @@ function ChartLegend<T>({
           </InternalBox>
         )}
 
-        <ul className={styles.list}>
+        <div className={styles.list}>
           {series.map((s, index) => {
             const someHighlighted = highlightedSeries !== null;
             const isHighlighted = highlightedSeries === s.datum;
             const isDimmed = someHighlighted && !isHighlighted;
             return (
-              <li
+              <div
+                {...focusVisible}
+                role="button"
                 key={index}
-                onMouseOver={handleMouseOver.bind(null, s.datum)}
-                onMouseLeave={handleMouseLeave}
                 className={clsx(styles.marker, {
                   [styles['marker--dimmed']]: isDimmed,
                   [styles['marker--highlighted']]: isHighlighted,
-                  [styles['marker--focused']]: isHighlighted && isFocused,
                 })}
-                aria-disabled={isDimmed ? true : undefined}
+                ref={elem => {
+                  if (elem) {
```

Comment:
would it be easier to assign the ref to the above element and use some selector e.g. `toolbarRef.current.querySelectorAll('div[role="button"]')[index]`?